FHEM/00_SIGNALduino.pm

```diff
@@ -2939,16 +2950,57 @@ sub SIGNALduino_FW_saveWhitelist
 		$wl_attr =~ s/,$//;			# Komma am Ende entfernen
 	}
 	$attr{$name}{whitelist_IDs} = $wl_attr;
-	SIGNALduino_Log3 $hash, 3, "$name Protocolist save: $wl_attr";
+	SIGNALduino_Log3 $name, 3, "$name Protocolist save: $wl_attr";
 	SIGNALduino_IdList("x:$name", $wl_attr);
 }
 
-sub SIGNALduino_FW_deselectAll
+sub SIGNALduino_FW_selectAll
```

Comment:
So auf den 1. Blick sieht das für mich wie eine ähnliche Funktion mit doppeltem Code wie bei `SIGNALduino_IdList` aus.

Hier werden aber nicht so viele Fälle abgedeckt. Können wir einen gemeinsamen code verwenden?

Reply:
Das sieht nur auf den 1. Blick wie eine ähnliche Funktion aus.
Für gemeinsamen Code ist es zu verschieden.

Reply:
Du möchstest also, daß wenn nach dem y eine Zahl kommt z.B. "y91" es nicht als Entwicklerversion erkannt wird.

Reply:
Ja, dann soll nur ein Protokoll aktiviert werden und nicht auf einmal alle.
Vielleicht wäre es hier gut kleine Funktionen zu erstellen die nur den Status für devFlag liefern.

Reply:
Das devFlag kann mit in die sub SIGNALduino_getAttrDevelopment rein
```
sub SIGNALduino_getAttrDevelopment
{
	my $name = shift;
	my $develop;
	my $devFlag = 0;
	if (index(SDUINO_VERSION, "dev") >= 0) {  	# development version
		$develop = AttrVal($name,"development", 0);
		$devFlag = 1 if ($develop eq "1" || (substr($develop,0,1) eq "y" && $develop !~ m/^y\d/)) {	# Entwicklerversion, y ist nur zur Abwaertskompatibilitaet und kann in einer der naechsten Versionen entfernt werden
	}
	else {
		$develop = "0";
		SIGNALduino_Log3 $name, 3, "$name IdList: ### Attribute development is in this version ignored ###";
	}
	return ($develop, $devFlag);
}
```